src/ng-select/lib/ng-select.component.ts

```diff
@@ -408,13 +409,19 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
     }
 
     select(item: NgOption) {
+        if (this.waitForSave) {
```

Comment:
Given that this returns unconditionally, I think it should not be called `waitForSave`. Maybe just call it `block` or something? It can obviously be used to wait for saving, but for other purposes as well 🏖

Reply:
🔧